spec/features/student_dashboard_view_spec.rb

```diff
@@ -0,0 +1,118 @@
+require 'rails_helper'
+
+describe 'viewing the dashboard as a student' do
+  let!(:student) { create :user, staff: false }
+  before :each do
+    when_current_user_is student, integration: true
+  end
+  context 'student has submitted an application' do
+    context 'student got an interview' do
+      let!(:interview) { create :interview, user: student }
+      before :each do
+        visit student_dashboard_url
+      end
+      it 'displays the date, time, and location of any interviews' do
+        expect page.has_text? interview.information
+      end
+      it 'contains a link to review the application' do
+        click_link 'Review your application',
+                   href: application_record_path(interview.application_record)
+        expect(page.current_url)
+          .to eql application_record_url(interview.application_record)
+      end
+    end # of student got an interview
+    context 'student did not get an interview' do
+      let!(:pending_application) do
+        create :application_record,
+               reviewed: false,
+               user: student
+      end
+      let!(:denied_application) do
+        create :application_record,
+               reviewed: true,
+               user: student,
+               staff_note: 'No'
+      end
+      before :each do
+        visit student_dashboard_url
+      end
+      context 'configured value notify of application denial reason is true' do
+        before :each do
+          allow_any_instance_of(ApplicationConfiguration)
```

Comment:
From the documentation:
"With a stub, if the message might be received with other args as well, you should stub a default value first, and then stub or mock the same message using with to constrain to specific arguments.

A message expectation will fail if the message is received with different arguments."